Add Namespaces

vinkla opened the pull request:
I've added namespaces and PSR-4 styled autoloading. As discussed in https://github.com/vinkla/laravel-pusher/issues/2 this pull request only adds namespaces. Everything else should work as before. This is a breaking change.

GrahamCampbell commented:
There are already namespaces on https://github.com/pusher/pusher-http-php/blob/new-lib.

vinkla commented:
> There are already namespaces on https://github.com/pusher/pusher-http-php/tree/new-lib.

Yes but that version hasn't been updated since April 2016. To my understanding @luismfonseca wanted a simple solution by adding namespaces for now.

luismfonseca commented:
Thanks for taking time doing this PR @vinkla. :)

I had a quick look around and it does seem to be doing more than what I wanted: which was just adding namespacing. I'd welcome separate PRs for other issues (e.g. .gitattributes or .editorconfig + style changes).

I would definetely prefer if the code styling was left untouched.

luismfonseca commented on src/Pusher.php:
```diff
@@ -131,9 +77,9 @@ public function __construct($auth_key, $secret, $app_id, $options = array(), $ho
-                $options['encrypted'] === true &&
-                !isset($options['scheme']) &&
-                !isset($options['port'])) {
+        $options['encrypted'] === true &&
```
I'm unsure about this stylistic change as then this reads almost as a separate instruction and not a continuation of the if statement.

luismfonseca commented on src/Pusher.php:
```diff
@@ -117,7 +63,7 @@ public function __construct($auth_key, $secret, $app_id, $options = array(), $ho
 
             $this->log('Legacy $host parameter provided: '.
-                                    $this->settings['scheme'].' host: '.$this->settings['host']);
+            $this->settings['scheme'].' host: '.$this->settings['host']);
```
I'm also a bit unsure about the style here, as it reads as a new statement and not a continuation of the previous one.

vinkla replied:
Yes, I agree. I'll fix this one.

vinkla replied:
Fixed this in https://github.com/pusher/pusher-http-php/pull/139/commits/da5aed654494a2e78b8374a0a006ba6b5c5c3978

luismfonseca replied:
I don't think that commit fixed this one. :o

vinkla replied:
Oh, okay. How do you want it to look?

luismfonseca commented:
I don't mind the moving of files from `lib` to `src` if that's the most standard way of doing libraries. It should be on a separate PR but I'm happy to keep this one this one.

luismfonseca commented on README.md:
````diff
@@ -55,12 +55,12 @@ The fourth parameter is an `$options` array. The additional options are:
 
 ```php
-$pusher = new Pusher( $app_key, $app_secret, $app_id, array( 'cluster' => $app_cluster, 'encrypted' => true ) );
+$pusher = new Pusher\Pusher( $app_key, $app_secret, $app_id, array( 'cluster' => $app_cluster, 'encrypted' => true ) );
````
I'm wondering how to update documentation... Should we suggest `new Pusher\Pusher` or should we say `use Pusher` to import this namespace and keep suggesting `new Pusher` here? (I'd appreciate your input on this as well @leesio).

vinkla replied:
In my opinion it doesn't matter. I think developers will understand either way.

vinkla commented:
> I had a quick look around and it does seem to be doing more than what I wanted: which was just adding namespacing. I'd welcome separate PRs for other issues (e.g. .gitattributes or .editorconfig + style changes).

These files doesn't affect the library in any way. The `.gitattributes` file help us remove unwanted files in production ([you can read more about it here](https://www.reddit.com/r/PHP/comments/2jzp6k/i_dont_need_your_tests_in_my_production/)). I've added the `.editorconfig` to make sure other developers also follow the indentation. Before, all lines didn't follow the same indentation.

> I would definetely prefer if the code styling was left untouched.

I've updated and fixed the indentation (4 spaces) in all files. This is recommended in the PSR-2 syntax standard: http://www.php-fig.org/psr/psr-2/#overview

> I don't mind the moving of files from lib to src if that's the most standard way of doing libraries. It should be on a separate PR but I'm happy to keep this one this one.

Great! As you've already stated, this is the most common naming convention.

Let me know if there is anything else I can do.

luismfonseca commented:
Alright, other than that cosmetic change, this LGTM.

I'll merge and release when other public documentation is ready to go.

vinkla commented:
> I'll merge and release when other public documentation is ready to go.

Great!
